streamz/core.py

```diff
@@ -991,6 +1013,27 @@ def __init__(self, *upstreams, **kwargs):
             self.emit_on = upstreams
         Stream.__init__(self, upstreams=upstreams, **kwargs)
 
+    def _add_upstream(self, upstream):
+        self.last.append(None)
+        self.missing.update([upstream])
+        if self.emit_on != self.upstreams:
```

Comment:
This equality check seems a little odd - can't we just store whether `emit_on` was originally given as `None`? Same comment for _remove_upstream.